docs.md

```diff
@@ -276,3 +276,26 @@ If `roletemplates.builtin` is true then all fields are immutable except:
  ### Deletion check
 
 RoleTemplate can not be deleted if they are referenced by other RoleTemplates via `roletemplates.roleTemplateNames` or by GlobalRoles via `globalRoles.inheritedClusterRoles`
+
+# provisioning.cattle.io/v1 
+
+## Cluster 
+
+### Validation Checks
+
+#### cluster.spec.clusterAgentDeploymentCustomization and cluster.spec.fleetAgentDeploymentCustomization
+
+`Key rule`: The keys are being validated to match the kubernetes requirement. 63 Characteres maximum, Regex:
+`([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]` it also accepts a "/"
+
+`appendTolerations`:  is being validated with the  `{Key Rule}` which live in [this apimachinery function](https://github.com/kubernetes/apimachinery/blob/02a41040d88da08de6765573ae2b1a51f424e1ca/pkg/apis/meta/v1/validation/validation.go#L96)
```

Comment:
Odd wording, maybe, '`appendTolerations`: is validated against the '{key Rule}' which lives in [this api machinery function](fakeurl)' ?